### ISSUE TITLE
axis clamping

### ISSUE THREAD

**Issue by mpdehaan** [sourceoptics/source_optics#115]
we should have a setting in django settings for GRAPH_CLAMP_FACTOR (default TBD) that clamps the Y axis of each (or most? - probably not needed for participation) graphs at some multiple of the mean.  

This will help anomalous spikes from making the graphs less usable without zooming in

**Comment by mpdehaan** [comment]
my thought is this setting should just be set on an organization as an org preference, with a default, to avoid the settings.py customization.

We should probably throw out values below, say, 10 (arbitrary), take the mean of that, multiply by maybe 5 or 10 or something, and then clamp at that point.  

But because a lot of people will have just 1 or 2 commits, we should not include those low numbers in our clamping selection algorithm.

**Comment by mpdehaan** [comment]
I pushed some basic support for graph clamping on the Y axis based on computing the Nth percentile score, seeing if the Nth percentile is some fraction from max, and then multiplying that Nth percentile by some other fraction.  This is all configurable and tweakable in settings.py.  The rough algorithm is in dataframes.py and can probably be improved, but seems to improve nearly all graphs.

I also reduced the graph granularity to "Monthly" if you select more than 700 or so weeks, which seems to help the look of the graphs quite a bit.  This should probably be a configurable number, so I'll do that next!
